packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/common/get-transaction-receipt-by-tx-id.ts

```diff
@@ -0,0 +1,203 @@
+import { LoggerProvider, LogLevelDesc } from "@hyperledger/cactus-common";
+import { Gateway } from "fabric-network";
+import { GetTransactionReceiptResponse } from "../generated/openapi/typescript-axios";
+import { common } from "fabric-protos";
+const { BlockDecoder } = require("fabric-common");
+export interface IGetTransactionReceiptByTxIDOptions {
+  readonly logLevel?: LogLevelDesc;
+  readonly gateway: Gateway;
+  readonly channelName: string;
+  readonly params: string[];
+}
+export async function getTransactionReceiptForLockContractByTxID(
+  req: IGetTransactionReceiptByTxIDOptions,
+): Promise<GetTransactionReceiptResponse> {
+  const fnTag = `getTransactionReceiptForLockContractByTxID`;
+  const log = LoggerProvider.getOrCreate({
+    label: fnTag,
+    level: req.logLevel || "INFO",
+  });
+  const { gateway } = req;
+
+  const contractName = "qscc";
+  const methodName = "GetBlockByTxID";
+  if (req.params.length != 2) {
+    throw new Error(`${fnTag}, should have 2 params`);
+  }
+  const network = await gateway.getNetwork(req.channelName);
+
+  const contract = network.getContract(contractName);
+  const out: Buffer = await contract.evaluateTransaction(
+    methodName,
+    ...req.params,
+  );
+  const reqTxID = req.params[1];
+  const block: common.Block = BlockDecoder.decode(out);
+  const blockJson = JSON.parse(JSON.stringify(block));
+  log.warn(blockJson);
+  log.warn("block number");
+  log.warn(blockJson.header.number);
+  //TODO: return block number
+  const transactReceipt: GetTransactionReceiptResponse = {};
+  transactReceipt.blockNumber = blockJson.header.number;
+  const txIDs = [];
+  if (block.data != undefined && block.data.data != undefined) {
```

Comment:
@jscode017 Please refactor it so that if the block cannot be found then it immediately bails with a `RuntimeError`. This will also allow you to narrow the types after that check.
In the endpoint class you have to check in the catch block if a user error happened or a genuine internal error. If it's a user error then an HTTP 400 should be sent instead of HTTP 500

Reply:
resolved